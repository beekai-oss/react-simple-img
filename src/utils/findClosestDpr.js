// @flow

const findClosestDpr = (result: Array<Object>, target: number) =>
  result.reduce((prev, curr) => (Math.abs(curr.dpr - target) < Math.abs(prev.dpr - target) ? curr : prev));

export default findClosestDpr;
