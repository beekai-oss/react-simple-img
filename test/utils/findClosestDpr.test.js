import findClosestDpr from '../../src/utils/findClosestDpr';

describe('findClosestDpr', () => {
  it('should find the closest dpr', () => {
    expect(findClosestDpr([
      { dpr: 1 },
      { dpr: 1.8 },
      { dpr: 1.2 },
    ], 2)).toMatchSnapshot();
  });
});
