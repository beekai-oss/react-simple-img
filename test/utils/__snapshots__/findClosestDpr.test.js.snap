// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findClosestDpr should find the closest dpr 1`] = `
Object {
  "dpr": 1.8,
}
`;
